### DISCUSSION TITLE
Homepage flash before showing child page when accessing direct link

### DISCUSSION THREAD

**Discussion by cramatt** [facebook/docusaurus#8291] [General]
When visiting a direct link in my site like example.com/docs/welcome I see the homepage flash briefly, then the child page loads. This is most notable on first visit or when hard refreshing. 

I thought this was maybe due to SSR or a config setting with addTrailingSlash. 

I am using nginx on production with proxy_pass. 

I don't see this on the actual [Doc](https://docusaurus.io/) website so I suspect this is due to my site configuration or nginx more likely. 

Any experience with this from the community?

**Comment by slorber** [comment]
hey

If you don't share the url where we can see the problem live it's impossible to troubleshoot this for you

The first thing I would try is to run your browser with JS disabled and see if the first render is good or if there's a problem in it. Most likely React fixes the problem after hydration
